draft-ietf-tls-tls13.md

```diff
@@ -5069,6 +5072,58 @@ lacking a "server_name" extension by terminating the connection with a
 "missing_extension" alert.
 
 
+## Protocol Invariants
+
+This section describes invariants that TLS endpoints and intermediaries MUST
+follow. It also applies to earlier versions, which assumed these rules but did
+not document them.
+
+TLS is designed to be securely and compatibly extensible. Newer clients or
+servers, when communicating with newer peers, should negotiate the most
+preferred common parameters with downgrade protection.  In particular, it is a
+security requirement (see {{security-handshake}}) that intermediaries passing
+traffic between a newer client and newer server without terminating TLS are
+unable to influence the handshake. At the same time, deployments update at
+different rates, so a newer client or server MAY continue to support older
+parameters, which would allow it to interoperate with older endpoints.
+
+For this to work, implementations MUST correctly handle extensible fields:
+
+- A client sending a ClientHello MUST support all parameters advertised in it.
+  Otherwise, the server may fail to interoperate by selecting one of those
+  parameters.
+
+- A server receiving a ClientHello MUST correctly ignore all unrecognized
+  cipher suites, extensions, and other parameters. Otherwise, it may fail to
+  interoperate with newer clients.
```

Comment:
In TLS 1.3 you need to add some text about the client handling NST and CR extensions.